OFEP-ofo-flag-service.md

```diff
@@ -0,0 +1,42 @@
+## flag service deployment driven by OpenFeature Operator
+
+## State: DRAFTING
+
+Currently, OpenFeature Operator (OFO) manages the deployment of a flag provider (e.g. flagd) by appending it to a pod's containers (sidecar pattern), thereby allowing containers within the pod to route to it. This OFEP campaigns for the extension of OFO to manage the deployment of an externally accessible flag provider (e.g. from client side applications, or even server side applications that don't want to use the sidecar pattern).
```

Comment:
Can we discuss how this would work in principal? Would it be limited to the same Kubernetes cluster or network segment? The first thought is around connectivity and the challenges there. Would say its the users responsibility to provide a flat network?

Reply:
Limiting the resources created to a Deployment & Service means that the networking beyond that is left as a responsibility to the users.
The deployment could be created in whichever namespace is provided but the Service CRUD is intentionally limited to OFO's namespace to minimise security risks.

In my mind, once the workload has been deployed (flagd) the Kubernetes primitives (i.e. Deployments) take charge of the lifecycle of the flag provider.

Reply:
Perhaps you could rephrase `This OFEP campaigns for the extension of OFO to manage the deployment of an externally accessible flag provider`  to something like `This OFEP campaigns for the extension of OFO to manage flag providers by abstracting a Kubernetes Deployment resource`.

Reply:
Great suggestion, done